src/rocksdb-types/src/config.rs

```diff
@@ -422,16 +470,75 @@ pub struct RocksDBConfig {
     pub stats_persist_interval_seconds: u32,
     pub point_lookup_block_cache_size_mb: Option<u32>,
     pub dynamic: RocksDBDynamicConfig,
+
+    /// Write buffer manager configs
+    pub write_buffer_manager_config: RocksDbWriteBufferManagerConfig,
+    /// Shared write buffer manager instance,
+    /// can only be instantiated once.
+    pub shared_write_buffer_manager: Arc<OnceCell<WriteBufferManager>>,
 }
 
-impl Default for RocksDBConfig {
-    fn default() -> Self {
-        RocksDBConfig::new(RocksDBTuningParameters::default())
+impl Debug for RocksDBConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RocksDBConfig")
+            .field("compaction_style", &self.compaction_style)
+            .field(
+                "optimize_compaction_memtable_budget",
+                &self.optimize_compaction_memtable_budget,
+            )
+            .field(
+                "level_compaction_dynamic_level_bytes",
+                &self.level_compaction_dynamic_level_bytes,
+            )
+            .field(
+                "universal_compaction_target_ratio",
+                &self.universal_compaction_target_ratio,
+            )
+            .field("parallelism", &self.parallelism)
+            .field("compression_type", &self.compression_type)
+            .field(
+                "bottommost_compression_type",
+                &self.bottommost_compression_type,
+            )
+            .field("retry_max_duration", &self.retry_max_duration)
+            .field(
+                "stats_log_interval_seconds",
+                &self.stats_log_interval_seconds,
+            )
+            .field(
+                "stats_persist_interval_seconds",
+                &self.stats_persist_interval_seconds,
+            )
+            .field(
+                "point_lookup_block_cache_size_mb",
+                &self.point_lookup_block_cache_size_mb,
+            )
+            .field("dynamic", &self.dynamic)
+            .field(
+                "write_buffer_manager_config",
+                &self.write_buffer_manager_config,
+            )
+            .finish_non_exhaustive() // skipping shared_write_buffer_manager
```

Comment:
OOC would it make sense to add a Debug implementation for `WriteBufferManager` in the rocksdb crate?

Reply:
this is fine!

Reply:
used derivative